src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -13,24 +16,35 @@ trait Producer[T] extends PipelineStep with CachingEnabled with Logging {
   self =>
   protected def create: T
 
+  /** Call `create` but store time taken. */
+  def createAndTime: T = {
+    val (result, duration) = Timing.time(this.create)
+    timing = Some(duration)
+    result
+  }
+
   /** Return the computed value. */
-  def get: T = {
-    if (!cachingEnabled)
-      create
-    else if (!initialized) {
+  final def get: T = {
+    if (!cachingEnabled) {
+      createAndTime
+    } else if (!initialized) {
       initialized = true
       cachedValue
-    }
-    else if (!cachedValue.isInstanceOf[Iterator[_]]) {
+    } else if (!cachedValue.isInstanceOf[Iterator[_]]) {
       cachedValue
-    }
-    else {
-      create
+    } else {
+      createAndTime
     }
   }
 
+  private var timing: Option[Duration] = None
   private var initialized = false
-  private lazy val cachedValue: T = create
+  private lazy val cachedValue: T = createAndTime
+
+  /** Report the amount of time taken in milliseconds, or None if the value is cached
+    * or this stage has not been run yet.
+    */
+  def timeTaken: Option[Duration] = timing
```

Comment:
I'm a little sad that, after taking the effort to move all those straggler fields from `Producer` to `PipelineStepInfo`, this field has crept back in.  But I think it makes sense, since this is not static information.